app/code/community/Bolt/Boltpay/Model/Admin/ExtraConfig.php

```diff
@@ -61,6 +61,13 @@ class Bolt_Boltpay_Model_Admin_ExtraConfig extends Mage_Core_Model_Config_Data
      */
     public function save()
     {
+        // when json_decode an empty string, PHP7 will trigger an Syntax error, so directly return without json_decode for empty string value
```

Comment:
We will still have the issue in function `getExtraConfig` lower down. Can we move this and the `json_decode` portion into `normalizeJSON`, rename it to `normalizeExtraConfig` or something, and make so we have a simple `$optionsJson = $this->normalizeExtraConfig($this->getValue); // Returns empty array if passed empty string`?

Reply:
@matt-thomason 

1. If the value is an empty string, it means the merchant want to remove any extra config setting, and based on current logic,  when $optionsJson = $this->normalizeExtraConfig($this->getValue);
whatever it get an empty array or string, the config value would be reset to old value, also it is hard to detect whether it is a json error or real value in such a case.

2. On the other hand, in the function `getExtraConfig`, if the Mage::getStoreConfig('payment/boltpay/extra_options') return an empty string, `$allExtraConfigs` would get an empty array, then $rawValue would be assigned an empty string too. So json_decode an empty string wont cause any exception or issue in this function.

Reply:
Sorry you are right - I missed this nuance.